pkg/dispatchable/dispatchable.go

```diff
@@ -0,0 +1,65 @@
+package dispatchable
+
+import (
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+func NewDispatchFinder(l hclog.Logger, sliceAtoms []types.Atom) *DispatchFinder {
+	mAtoms := make(map[types.SpecTuple]types.Atom)
+	for _, atom := range sliceAtoms {
+		mAtoms[atom.Spec] = atom
+	}
+	x := DispatchFinder{
+		l:     l,
+		atoms: mAtoms,
+	}
+	return &x
+}
+
+// IsDispatchable determines whether a specific package could be dispatched
+// right now.
+func (d *DispatchFinder) IsDispatchable(spec types.SpecTuple, p *types.Package) bool {
+	hAtom := d.atoms[types.SpecTuple{spec.Host, spec.Host}]
+	for hdep, _ := range p.HostDepends {
+		if hAtom.Pkgs[hdep].Dirty {
+			return false
+		}
+	}
+
+	tAtom := d.atoms[spec]
+	for dep, _ := range p.MakeDepends {
+		if tAtom.Pkgs[dep].Dirty {
+			return false
+		}
+	}
+	for dep, _ := range p.Depends {
+		if tAtom.Pkgs[dep].Dirty {
+			return false
+		}
+	}
+	// If we get this far, all hostdeps, makedeps, deps are clean.
+	return true
+}
+
+// ImmediatelyDispatchable returns a map of tuples -> packages that can be
+// hypothetically dispatched right now.
+// *Assumes graph is freshly Cleaned*.
```

Comment:
What is the penalty if it isn't?

Reply:
Will return packages that are already clean.. for clarity, adding to comment.